keps/sig-node/2369-sizing-providers/README.md

```diff
@@ -0,0 +1,319 @@
+// image provided in the original request. Kubelet will read the response from the plugin via stdout.
+type SizingProviderResponse struct {
+	metav1.TypeMeta
+ 
+      // A set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=150G,pid=100) pairs
+	// that describe resources reserved for non-kubernetes components.
+	// Currently only cpu and memory are supported.
+	// See http://kubernetes.io/docs/user-guide/compute-resources for more detail.
+	SystemReserved map[string]string `json:"systemReserve"`
+	// A set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=150G,pid=100) pairs
+	// that describe resources reserved for kubernetes system components.
+	// Currently cpu, memory and local ephemeral storage for root file system are supported.
+	// See http://kubernetes.io/docs/user-guide/compute-resources for more detail.
+	KubeReserved map[string]string `json:"kubeReserve"`
+}
+
+```
+
+### Test Plan
+
+Alpha:
+* unit tests for the exec plugin provider
+* unit tests for API validation
+
+### Graduation Criteria
+
+### Alpha
+
+* adequate unit testing for the plugin provider
+* a working reference implementation, proving that the existing functionality of the built-in providers
+can be achieved using the exec plugin.
+
+#### Alpha -> Beta Graduation
+
+* integration or e2e tests.
+* at least one working plugin implementation.
+
+#### Beta -> GA Graduation
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+This feature is feature gated so explicit opt-in is required on upgrade and explicit opt-out is required on downgrade.
+
+### Version Skew Strategy
+
+Not applicable because this feature is contained to only the kubelet and does not require communication to other components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: KubeletSizingProvider
+    - Components depending on the feature gate: kubelet
+
+* **Does enabling the feature change any default behavior?**
+  No, use of this feature still requires extra flags enabled on the kubelet.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, as long as kubelet does not specify the flags `--sizing-provider-config` and `--sizing-provider-bin-dir`.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  Kubelet will continue to invoke exec plugins. No state is stored for this feature to function.
+
+* **Are there any tests for feature enablement/disablement?**
+  Yes.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+TBD for beta.
+
+* **What specific metrics should inform a rollback?**
```

Comment:
This isn't strictly required for alpha, but as I recall, if these values are too low it can lead to node readiness/responsiveness problems.  It may be a good idea to indicate how to tell if the current size is well suited to the current usage.

Reply:
Thanks @deads2k for your comment.